### PR TITLE
Adds throw error command in Go client

### LINKED ISSUE #3507: I can throw a job error from the Go client
**Is your feature request related to a problem? Please describe.**
Using the Go client, I want to throw an error if a business problem (i.e. non-technical) occurs while processing a job. The job should not be retried. Instead, the error should be handled by the workflow instance.

**Describe the solution you'd like**
* new API to throw an error for a particular job
* request parameters: 
  * job key - the key of the corresponding job
  * error code (any string) - must match to the error code defined in the workflow
  * error message (optional) - additional context, will be part of the incident if the error is not caught
* empty response 

**Describe alternatives you've considered**
none

**Additional context**
Related issues:
#3497 
#3498 
#3506

### PR DESCRIPTION
**Pull request by miguelpires**
## Description

The first commit:
* add a ThrowError command to the protocol
* implements the command 
* updates the README because the mockgen command was not updating the grpc interface mocks

The second commit refactors the commands to eliminate some duplicated/unnecessary code and make initialization simpler.

## Related issues

closes #3507

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: clients/go/pkg/commands/throwError_command.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+type ThrowErrorCommandStep1 interface {
+	JobKey(int64) ThrowErrorCommandStep2
+}
+
+type ThrowErrorCommandStep2 interface {
+	ErrorCode(string) ThrowErrorCommandStep3
+}
+
+type ThrowErrorCommandStep3 interface {
+	ErrorMessage(string) DispatchThrowErrorCommand
```
Comment (saig0) [inline clients/go/pkg/commands/throwError_command.go]: The error message should be optional. I'm not sure but it seems that it is currently required, right?

File: gateway-protocol/src/main/proto/gateway.proto [hunk 2]
```diff
@@ -397,6 +409,19 @@ service Gateway {
   rpc FailJob (FailJobRequest) returns (FailJobResponse) {
   }
 
+  /*
+    Throws an error . If an error catch event with the specified errorCode can't be found, an incident will be raised instead.
```
Comment (saig0) [inline gateway-protocol/src/main/proto/gateway.proto]:
```suggestion
    Reports a business error (i.e. non-technical) that occurs while processing a job. The error is handled in the workflow by an error catch event. If there is no error catch event with the specified errorCode then an incident will be raised instead.
```

**Review by saig0 — CHANGES_REQUESTED** [review]
Looks good :+1:  (as far as I can tell :sweat_smile:)

Just a minor comment regarding the error message.

**Comment by MiguelPires** [comment]
Thanks for the feedback @saig0. I addressed those issues

**Review by saig0 — APPROVED** [review]
:+1:

**Pull request merged**